src/Select/SelectHelper.php

```diff
@@ -0,0 +1,160 @@
+                    $selectColumns = array_merge($selectColumns, $selectFields);
+                    $foundSelect = true;
+                }
+
+                foreach (self::DIRECTIVES as $directiveType) {
+                    if ($directive = ASTHelper::directiveDefinition($fieldDefinition, $directiveType)) {
+                        assert($directive instanceof DirectiveNode);
+
+                        $relationName = ASTHelper::directiveArgValue($directive, 'relation', $field);
+                        if (method_exists($model, $relationName)) {
+                            $relation = $model->{$relationName}();
+                            if (in_array($directiveType, self::DIRECTIVES_REQUIRING_LOCAL_KEY)) {
+                                $selectColumns[] = self::getLocalKey($relation);
+                            } elseif (in_array($directiveType, self::DIRECTIVES_REQUIRING_FOREIGN_KEY)) {
+                                $selectColumns[] = self::getForeignKey($relation);
+                            } elseif (in_array($directiveType, self::DIRECTIVES_REQUIRING_MORPH_KEY)) {
+                                $selectColumns[] = self::getForeignKey($relation);
+                                $selectColumns[] = $relation->getMorphType();
+                            } else {
+                                $selectColumns[] = $model->getKeyName();
+                            }
+                        }
+
+                        continue 2;
+                    }
+                }
+
+                if ($foundSelect) {
+                    continue;
+                }
+                if ($directive = ASTHelper::directiveDefinition($fieldDefinition, 'rename')) {
+                    // append renamed attribute to selection
+                    $renamedAttribute = ASTHelper::directiveArgValue($directive, 'attribute');
+                    $selectColumns[] = $renamedAttribute;
+                } elseif (($directive = ASTHelper::directiveDefinition($fieldDefinition, 'method')) || method_exists($model, $field)) {
+                    $relation = null !== $directive
+                        ? $model->{ASTHelper::directiveArgValue($directive, 'name')}()
+                        : $model->{$field}();
+                    if ($relation instanceof MorphTo) {
+                        $selectColumns[] = self::getForeignKey($relation);
+                        $selectColumns[] = $relation->getMorphType();
+                    } elseif ($relation instanceof BelongsTo) {
+                        $selectColumns[] = self::getForeignKey($relation);
+                    } elseif ($relation instanceof HasOneOrMany) {
+                        $selectColumns[] = self::getLocalKey($relation);
+                    } else {
+                        $selectColumns[] = $model->getKeyName();
+                    }
+                } else {
+                    // fallback to selecting the field name
+                    $selectColumns[] = $field;
```

Comment:
This just goes wrong for the very simple case of there being a custom getter for the field - or any custom directive that we can not possibly know about. Consider the following:

```graphql
type MyModel {
  foo: Int @someCustomDirectiveThatWeDoNotKnowAbout
  bar: ID @field(resolver: "SomeCustomClass@andAMethodWeCanNotPossiblyLookInto")
}
```

I don't think any amount of magic can help us here, any approach of trying to determine columns without explicit configuration is just fundamentally flawed.